src/utils/useChatHandler.ts

```diff
@@ -0,0 +1,54 @@
+import { useState } from "react"
+import { useChatHistory } from "./useChatHistory"
+import { ChatAPI } from "@/apis/chat"
+
+export const useChatHandler = (botId: string, CHAT_SESSION_ID: string) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [isError, setIsError] = useState<boolean>(false)
+
+  // Vu: Custom hook to manage chat history.
+  /* 
+Vu: in the old commit, I use useChatHistory directly in ChatWindow.
+- In ChatWindow, it needs chatHistory and clearChatHistory from useChatHistory
+- In this useChatHandler, it needs addMessageToChatHistory from useChatHistory
+- If I declare both useChatHistory in both ChatWindow and useChatHandler, it will create 2 separated instances in both places. Resulting an error: when user chat, it doesn't show in UI although the server responded and have new data in Session Storage.
+- That's why we include useChatHistory inside this hook, and return all useChatHistory's values so that they can be used in ChatWindow.
+  */
+  const { chatHistory, addMessageToChatHistory, clearChatHistory } =
+    useChatHistory()
+
+  // Vu: This function display new message (from message input and clickable options) to UI, also send data to server and display answer in UI
+  const sendMessageToServerAndDisplay = async (messageFrom: string) => {
+    addMessageToChatHistory(messageFrom, "user")
+
+    // Send the message to the server
+    try {
+      setIsLoading(true)
+      const response = await ChatAPI.send({
+        botId,
+        chatSessionId: CHAT_SESSION_ID,
+        command: messageFrom,
+      })
+
+      // Handle edge cases such as when frontend doesn't provide enough data to server
+      if (!response.output) throw Error
```

Comment:
can we setError(true) here?

Reply:
<img width="746" alt="image" src="https://github.com/AI-ChatBot-Yoga/ai-chatbot/assets/133303660/000463a6-9600-4738-8365-ac08d844354b">
May I ask why do we set `setError(true)` after throw an error, since it is going to `catch` block and it has `setError(true)` there.

Reply:
@vuson1709 I tried your code on my local and it just throw an error in the console and didn't show the error message on the UI 🤔 Does this happen on your side?

Reply:
No it doesn't. Please go to Application -> Session storage and delete everything in here 

<img width="754" alt="image" src="https://github.com/AI-ChatBot-Yoga/ai-chatbot/assets/133303660/15e6464c-b882-48af-81ed-ca83957a12a2">

Reply:
I tried to un-comment ` chatSessionId: CHAT_SESSION_ID,` in
```javascript
    // Send the message to the server
    try {
      setIsLoading(true)
      const response = await ChatAPI.send({
        botId,
        // chatSessionId: CHAT_SESSION_ID,
        command: messageFrom,
      })
``` 

then test the chat, it still displays the error. Also I move the error from bottom of the chat to the top to solve `Error text overlap with chat bubble` that we discussed in QA session
<img width="329" alt="image" src="https://github.com/AI-ChatBot-Yoga/ai-chatbot/assets/133303660/347a27ad-fa3f-49c9-811a-94f525eb754f">

Reply:
<img width="466" alt="image" src="https://github.com/AI-ChatBot-Yoga/ai-chatbot/assets/80876101/409d1a8a-7fe8-4883-a036-484285dc30ea">
 it works now, thank you! 🙏